iodata/test/test_mwfn.py

```diff
@@ -0,0 +1,196 @@
+    assert_equal(np.sum(mol.mo.occsb), 4.0)
+    assert_equal(mol.energy, -3.90732095E+01)
+    assert_allclose([shell.angmoms[0] for shell in mol.obasis.shells], [0, 0, 1, 0, 0, 0])
+    assert_allclose([shell.icenter for shell in mol.obasis.shells], [0, 0, 0, 1, 2, 3])
+    assert_allclose([shell.nprim for shell in mol.obasis.shells], [3, 3, 3, 3, 3, 3])
+    exponents1 = np.array([7.16168373E+01, 1.30450963E+01, 3.53051216E+00])
+    exponents2 = np.array([2.94124936E+00, 6.83483096E-01, 2.22289916E-01])
+    exponents3 = np.array([2.94124936E+00, 6.83483096E-01, 2.22289916E-01])
+    exponents4 = np.array([3.42525091E+00, 6.23913730E-01, 1.68855404E-01])
+    assert_allclose(mol.obasis.shells[0].exponents, exponents1)
+    assert_allclose(mol.obasis.shells[1].exponents, exponents2)
+    assert_allclose(mol.obasis.shells[2].exponents, exponents3)
+    assert_allclose(mol.obasis.shells[3].exponents, exponents4)
+    assert_allclose(mol.obasis.shells[4].exponents, exponents4)
+    assert_allclose(mol.obasis.shells[5].exponents, exponents4)
+    coeffs1 = np.array([[1.54328967E-01], [5.35328142E-01], [4.44634542E-01]])
+    coeffs2 = np.array([[-9.99672292E-02], [3.99512826E-01], [7.00115469E-01]])
+    coeffs3 = np.array([[1.55916275E-01], [6.07683719E-01], [3.91957393E-01]])
+    coeffs4 = np.array([[1.54328967E-01], [5.35328142E-01], [4.44634542E-01]])
+    assert_allclose(mol.obasis.shells[0].coeffs, coeffs1)
+    assert_allclose(mol.obasis.shells[1].coeffs, coeffs2)
+    assert_allclose(mol.obasis.shells[2].coeffs, coeffs3)
+    assert_allclose(mol.obasis.shells[3].coeffs, coeffs4)
+    assert_allclose(mol.obasis.shells[4].coeffs, coeffs4)
+    assert_allclose(mol.obasis.shells[5].coeffs, coeffs4)
+    # test first molecular orbital information
+    coeff = np.array([9.92532359E-01, 3.42148679E-02, 3.30477771E-06, - 1.97321450E-03,
+                      0.00000000E+00, -6.94439001E-03, - 6.94439001E-03, - 6.94539905E-03])
+    assert_equal(mol.mo.coeffs[:, 0], coeff)
+    mo_energies = np.array([-1.09902284E+01, -8.36918686E-01, -5.24254982E-01, -5.23802785E-01,
+                            -1.26686819E-02, 6.64707810E-01, 7.68278159E-01, 7.69362712E-01])
+    assert_allclose(mol.mo.energies, mo_energies)
+    assert_equal(mol.mo.occs[0], 2.000000)
+    assert_equal(mol.extra['mo_type'][0], 0)
+    assert_equal(mol.extra['mo_sym'][0], '?')
+    # test that for the same molecule fchk and mwfn generate the same objects.
+    olp = compute_overlap(mol.obasis, mol.atcoords)
+    mol2 = load_format_helper('ch3_rohf_sto3g_g03.fchk')
+    olp_fchk = compute_overlap(mol2.obasis, mol2.atcoords)
+    assert_allclose(mol.atcoords, mol2.atcoords, atol=1E-7, rtol=1E-7)
+    assert_allclose(mol2.obasis.shells[0].coeffs, coeffs1)
+    # fails due to special SP basis storage (leaving commented lines in for future reference)
+    # assert_allclose(mol2.obasis.shells[1].coeffs, coeffs2)
+    # assert_allclose(mol2.obasis.shells[2].coeffs, coeffs3)
```

Comment:
This should be fixable. Something like:
```suggestion
    # Mind the SP contraction...
    assert_allclose(mol2.obasis.shells[1].coeffs[:, 0], coeffs2)
    assert_allclose(mol2.obasis.shells[1].coeffs[:, 1], coeffs3)
    assert_allclose(mol2.obasis.shells[2].coeffs, coeffs4)
```
(may need to be further corrected though.)

Reply:
assert_allclose(mol2.obasis.shells[1].coeffs[:, 0], np.squeeze(coeffs2.T)) works well enough